File: app/src/main/java/com/ragdroid/dahaka/user/BaseUserActivity.java
package com.ragdroid.dahaka.user;

import android.os.Bundle;
import android.support.annotation.Nullable;

import com.ragdroid.dahaka.DahakaApplication;
import com.ragdroid.dahaka.app.UserManager;
import com.ragdroid.dahaka.mvp.BaseActivity;
import com.ragdroid.dahaka.mvp.BasePresenter;

import javax.inject.Inject;

import dagger.android.AndroidInjection;


/**
 * Created by garimajain on 18/08/17.
 */

public abstract class BaseUserActivity<T extends BasePresenter> extends BaseActivity<T> {

    @Inject UserManager userManager;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (!userManager.isLoggedIn()) {
            finishView();
        }
    }

    @Override
    protected void androidInject() {
        AndroidInjection.inject(this);
    }

    protected void logoutUser() {
        userManager.logOut();
        finishView();
    }


}

File: app/src/main/java/com/ragdroid/dahaka/user/BaseUserActivity.kt
package com.ragdroid.dahaka.user

import android.os.Bundle

import com.ragdroid.dahaka.DahakaApplication
import com.ragdroid.dahaka.app.UserManager
import com.ragdroid.dahaka.mvp.BaseActivity
import com.ragdroid.dahaka.mvp.BasePresenter
import com.ragdroid.dahaka.mvp.BaseView


/**
 * Created by garimajain on 18/08/17.
 */

abstract class BaseUserActivity<T : BasePresenter<V>, V : BaseView> : BaseActivity<T, V>() {

    internal lateinit var userManager: UserManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (!userManager.isLoggedIn) {
            finishView()
        }
    }

    override fun androidInject() {
        userManager = (application as DahakaApplication)
                .appComponent.userManager
        UserInjection.inject(this, userManager)
    }

    protected fun logoutUser() {
        userManager.logOut()
        finishView()
    }


}
